frontend/src/concepts/pipelines/content/createRun/contentSections/ParamsSection/IntegerInputParam.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { NumberInput } from '@patternfly/react-core';
+import { InputParamProps } from './types';
+
+export const IntegerInputParam: React.FC<InputParamProps> = ({ id, value, onChange }) => {
+  const [integerValue, setIntegerValue] = React.useState<number | ''>(parseInt(String(value)) || 0);
+
+  const onMinus = (event: React.MouseEvent<Element>) => {
+    const newValue = (integerValue || 0) - 1;
+    setIntegerValue(newValue);
+    onChange(event, newValue);
+  };
+
+  const onPlus = (event: React.MouseEvent<Element>) => {
+    const newValue = (integerValue || 0) + 1;
+    setIntegerValue(newValue);
+    onChange(event, newValue);
+  };
+
+  return (
+    <NumberInput
```

Comment:
We have a NumberInputWrapper component that handles a lot of the logic you did in this function.

Reply:
this should also help to not need to use `as HTMLInputElement` in the onChange